docs(breadcrumb, checkbox, clearbutton, coachindicator): Site Docs to Storybook (#2793)
adobe/spectrum-css: components/breadcrumb/stories/breadcrumb.mdx

import { Canvas, ArgTypes, Meta, Description, Title } from '@storybook/blocks';

import * as BreadcrumbStories from './breadcrumb.stories';

<Meta of={BreadcrumbStories} title="Docs" />

<Title of={BreadcrumbStories} />
<Description of={BreadcrumbStories} />

### Standard

<Canvas of={BreadcrumbStories.Default} />

### Dragged
<Canvas of={BreadcrumbStories.Dragged} />

### Nested
<Canvas of={BreadcrumbStories.Nested} />

### Nested (root visible)
<Canvas of={BreadcrumbStories.NestedRootVisible} />

### Multiline
<Canvas of={BreadcrumbStories.Multiline} />

### Multiline (dragged)
<Canvas of={BreadcrumbStories.MultilineDragged} />

### Multiline Nested

> <Canvas of={BreadcrumbStories.NestedMultiline} />

Author: For each story, I used whatever heading was on the docs site. But those names don't always align with previously written stories (for example, the heading on the docs site is "Standard," but the story name is "Default." My view for this first batch of work was to just match what's on the docs site, even if it was a little inconsistent, but I'm open to changing things to make them consistent.

import { Canvas, ArgTypes, Meta, Description, Title } from '@storybook/blocks';

import * as BreadcrumbStories from './breadcrumb.stories';

<Meta of={BreadcrumbStories} title="Docs" />

<Title of={BreadcrumbStories} />
<Description of={BreadcrumbStories} />

### Standard

<Canvas of={BreadcrumbStories.Default} />

### Dragged

<Canvas of={BreadcrumbStories.Dragged} />

### Nested

<Canvas of={BreadcrumbStories.Nested} />

### Nested (root visible)

<Canvas of={BreadcrumbStories.NestedRootVisible} />

### Multiline

<Canvas of={BreadcrumbStories.Multiline} />

### Multiline (dragged)

<Canvas of={BreadcrumbStories.MultilineDragged} />

> ### Multiline Nested

Author: I love this, that's a really great point about the first-time visitors, added it in here: https://github.com/adobe/spectrum-css/pull/2793/commits/8dd9fca301b60977c187de2406cd2cdd26adbd60

However, in trying to work that in, I may have found a bug for the `is-disabled` state. It seems like that class is added to the `.spectrum-Breadcrumbs-item`/parent `li`, instead of `.spectrum-Breadcrumbs-itemLink`. We had a block for disabled styles for `.spectrum-Breadcrumbs-itemLink &.is-disabled`, but that's not the behavior I was getting. [This fix](https://github.com/adobe/spectrum-css/pull/2793/commits/27b5ef2f056948b42247cdcefc6871c24f3a551e) seems to do what I'm expecting, but am I understanding things all the way? Am I missing anything? Should I be declaring `isDisabled` somewhere else in the components?

> ### Multiline Nested

Author: per our slack conversation about fixing where the `is-disabled` is being applied in the template, here's the relevant commits! 

[fix the template](https://github.com/adobe/spectrum-css/pull/2793/commits/d9702da4cbbc85d053b949c261711f0aed10df06)
[add disabled story to mdx with is-disabled notes](https://github.com/adobe/spectrum-css/pull/2793/commits/a7dbbb9337f2f7e34908c0220c0375925032ff54)

> <Canvas of={BreadcrumbStories.NestedMultiline} />

Author: For each story, I used whatever heading was on the docs site. But those names don't always align with previously written stories (for example, the heading on the docs site is "Standard," but the story name is "Default." My view for this first batch of work was to just match what's on the docs site, even if it was a little inconsistent, but I'm open to changing things to make them consistent.

<Meta of={BreadcrumbStories} title="Docs" />

<Title of={BreadcrumbStories} />
<Description of={BreadcrumbStories} />

## Variants

### Standard

<Canvas of={BreadcrumbStories.Default} />

### Dragged

<Canvas of={BreadcrumbStories.Dragged} />

### Nested

<Canvas of={BreadcrumbStories.Nested} />

### Nested (root visible)

<Canvas of={BreadcrumbStories.NestedRootVisible} />

### Multiline

<Canvas of={BreadcrumbStories.Multiline} />

### Multiline (dragged)

<Canvas of={BreadcrumbStories.MultilineDragged} />

### Multiline Nested

<Canvas of={BreadcrumbStories.NestedMultiline} />

### Multiline Nested (root visible)

<Canvas of={BreadcrumbStories.MultilineNestedRootVisible} />

### Compact

<Canvas of={BreadcrumbStories.Compact} />

### Compact Nested

<Canvas of={BreadcrumbStories.CompactNested} />

### Compact Nested (root visible)

<Canvas of={BreadcrumbStories.CompactNestedRootVisible} />

### Disabled

The example below has two disabled breadcrumb items. When disabling the text link, the `is-disabled` class gets added to `.spectrum-Breadcrumbs-itemLink`. When disabling the Action button, the `[disabled]` attribute is applied.

<Canvas of={BreadcrumbStories.Disabled} />

> ## Properties

Reviewer: ```suggestion
## Properties

The component accepts the following inputs (properties):
```

From the .storybook/DocumentationTemplate.mdx template that adds a nice context.

Author: Oh great idea. I might make a card to go back and double check some of the other recently-merged components and make sure they have this as well.

Author: should be added in this commit: https://github.com/adobe/spectrum-css/pull/2793/commits/56a86b9a361961f2ee6757764347e16f3523d730

Reviewer 2: @castastrophe Should we name this heading "Story arguments", "Story args", "Story arg types" or something like that, instead of "Properties"? I'm wondering if "Properties" if the best naming for what these are and what Storybook calls them. This would apply to our auto docs template and custom MDX files.

Reviewer: Let's leave properties for now. Yes, we're expressing them in storybook as args but consumers reading the docs shouldn't need to understand how storybook works to understand what it is.  In this case, this is documenting the design API.

> The example below has two disabled breadcrumb items. Any breadcrumb item can be disabled by adding the `is-disabled` class. When targeting a link, the `is-disabled` class gets added to `.spectrum-Breadcrumbs-itemLink`. When used on a button with the `.spectrum-ActionButton`, the `[disabled]` attribute is applied in addition to the `is-disabled` class.

Reviewer 2: ```suggestion
The example below has two disabled breadcrumb items. When disabling the text link, the `is-disabled` class gets added to `.spectrum-Breadcrumbs-itemLink`. When disabling the Action button, the `[disabled]` attribute is applied.
```
